### ISSUE TITLE
Unauthorized credential for client UID=0 GID=0

### ISSUE THREAD

**Issue by ZXRobotum** [dun/munge#130]
Hello all,

I somehow just have a problem with MUNGE in conjunction with SLURM with the error message shown in the headline and I just don't have a "PLAN" why I get this error message.

It works fine with my "small" cluster, but not with the productive cluster. I have to say that I am currently still using the MUNGE from DEBIAN as a distribution... 

The "munge.key" is the same everywhere and I do not get any error message when calling "munge -n | unmunge" or "ssh <host> "munge -c0 -z0 -n" | unmunge. Only in connection with SLURM. According to SLURM, the connection is accepted, but a "feedback" is then no longer given....

Can anyone help me with this?
Thanks in advance.....

Z. Matthias

**Comment by ZXRobotum** [comment]
Thank you very much for your help.
On Slurm.conf is have these:
SlurmUser=slurm
SlurmdUser=root

Well, I create the new "munge.key" with the following command, like your page:
sudo -u munge ${sbindir}/mungekey --verbose

On all my systems UID & GID from slurs & munge are the same....

As I wrote before, my small test cluster works fine with the same settings, compile operations, etc. Only the large cluster does not like more... And I don't find these mistake, failure, etc.

**Comment by dun** [comment]
The `munge.key` appears to be fine since the credential is successfully decoded. But the credential has been encoded with a uid restriction for a non-root user (presumably the _slurm_ user), and the process attempting to decode it is running as _root_ (hence the authorization error).

The only advice I have to offer here is to double-check that the _slurm_ account (with the same uid) exists on all nodes in your large cluster, that `SlurmUser=slurm` is set, and perhaps restart the _slurm_ service on all nodes in case the configuration changed after the service was initially started.

**Comment by tazend** [comment]
Hi @ZXRobotum 

just came accross this error too in our productive cluster and wondered why this happens.
Do the productive and your test cluster have the same slurm version?

Because in Slurm 22.05 (presumably) the Slurm devs added [this check](https://github.com/SchedMD/slurm/blob/slurm-22.05/src/plugins/auth/munge/auth_munge.c#L138) in the `init` function of their `auth_munge` plugin.

Now, whenever a job starts, a new `slurmstepd` process is spawned and this `init` function is called. If you have verbose mode on, you should see something like this very often in your syslog:

```
slurmstepd[16784]: cred/munge: init: Munge credential signature plugin loaded
```

As the comments in their source code say, they only check if `munge` is configured to allow the `root` user to decode any  incoming credential. They are creating a pseudo credential encoded with a different uid, and try to decode this as the `root` User. If this succeeds, you would be prompted with a error-message telling you to disable the setting in `munge` that `root` is allowed to decode any credential, as only the `SlurmUser` should be allowed to decode the credentials from `slurmctld`.

In short, the `Unauthorized credential for client UID=0 GID=0` is just a byproduct of their test in this `init` function and can be, as to my (hopefully correct) interpretation, savely ignored.

**Comment by dun** [comment]
@tazend, thanks for looking into this! :star:

It's unfortunate their safety check is causing confusion. Ideally this would be documented in their installation faq.

**Comment by tazend** [comment]
Hi @ZXRobotum 

We can continue discussion here in [this issue](https://github.com/tazend/me/issues/1), if you want (just to not further hijack this issue for discussing perhaps unrelated slurm errors.)

**Comment by tazend** [comment]
@dun 

Yeah a mention in the documentation would be good. I opened a bug report for this: https://bugs.schedmd.com/show_bug.cgi?id=16035
Lets see what they say :)
